doc/source/mergingdata.rst

```diff
@@ -117,18 +117,186 @@ Note more complicated spatial relationships can be studied by combining geometri
 Sjoin Performance
 ~~~~~~~~~~~~~~~~~~
 
-Existing spatial indexes on either `left_df` or `right_df` will be reused when performing an ``sjoin``. If neither df has a spatial index, a spatial index will be generated for the longer df. If both have a spatial index, the `right_df`'s index will be used preferentially. Performance of multiple sjoins in a row involving a common GeoDataFrame may be improved by pre-generating the spatial index of the common GeoDataFrame prior to performing sjoins using ``df1.sindex``.
+Most of the computation is done within the spatial index queries. The spatial index is always queried as follows:
 
 .. code-block:: python
 
-    df1 = # a GeoDataFrame with data
-    df2 = # a second GeoDataFrame
-    df3 = # a third GeoDataFrame
+   right_df.sindex.query_bulk(left_df.geometry, op)
 
-    # pre-generate sindex on df1 if it doesn't already exist
-    df1.sindex
+This translates to:
+
+.. code-block:: python
+
+   geom_in_left_df.op(geom_in_right_df)
```

Comment:
I don't think we need to explain internals of the implementation. User Guide should focus on _how do I make a spatial join_, not the explanation what we do under the hood. The existing note on performance also only guides you towards more performant use.

Reply:
Addressed.